src/helpers/stake/magicGlp/subgraph/getChartData/getGlpData.ts

```diff
@@ -0,0 +1,134 @@
+import axios from "axios";
+import { sortBy } from "lodash";
+
+const urls = {
+  42161: "gmx-stats",
+  43114: "gmx-avalanche-stats",
+};
+
+const fillNa = (arr: any) => {
+  const prevValues: any = {};
+  let keys: any;
+  if (arr.length > 0) {
+    keys = Object.keys(arr[0]);
+    delete keys.timestamp;
+    delete keys.id;
+  }
+
+  for (const el of arr) {
+    for (const key of keys) {
+      if (!el[key]) {
+        if (prevValues[key as keyof typeof prevValues]) {
+          el[key] = prevValues[key as keyof typeof prevValues];
+        }
+      } else {
+        prevValues[key] = el[key];
+      }
+    }
+  }
+  return arr;
+};
+
+export const getGlpData = async ({ from, to, chainId = 42161 }: any) => {
+  const url = urls[chainId as keyof typeof urls];
+  const subgraphUrl = `https://api.thegraph.com/subgraphs/name/gmx-io/${url}`;
+  const timestampProp = chainId === 42161 ? "id" : "timestamp";
```

Comment:
have a config for this instead of the specific "if" and store per-chain config:
```
myConfig: {
   42161: "id"
}
```

and then something like ```timestampProp = myConfig[chainId] || "timestamp"```